src/components/node-list/node-list-row.js

```diff
@@ -57,11 +58,72 @@ const NodeListRow = memo(
     visibleIcon = VisibleIcon,
     invisibleIcon = InvisibleIcon,
     rowType,
+    focusMode,
+    parentDisabled,
+    parentPipeline,
   }) => {
-    const VisibilityIcon = checked ? visibleIcon : invisibleIcon;
+    const VisibilityIcon =
+      type === 'modularPipeline'
+        ? FocusModeIcon
+        : checked
+        ? visibleIcon
+        : invisibleIcon;
     const isButton = onClick && kind !== 'filter';
     const TextButton = isButton ? 'button' : 'div';
 
+    const determineFocusMode = (focusMode, type, id) =>
+      focusMode !== null && type === 'modularPipeline' && id === focusMode?.id;
+
+    const determineDisabledLabel = (
+      parentPipeline,
+      parentDisabled,
+      applyFocusMode,
+      disabled
+    ) => {
+      if (parentPipeline === 'main') {
+        return disabled;
+      }
+      return (
+        parentDisabled !== false &&
```

Comment:
yes it is - we are checking that `parentDisabled` is not `true` or `undefined` ( which happens in the case of rows representing nodes instead of modular pipelines). The parentDisabled value is assigned `false` by default for modular pipelines hence using this check allows us to cater for both the `true` or `undefined` cases as well.